newIDE/app/src/VariablesList/index.js

```diff
@@ -187,9 +196,29 @@ export default class VariablesList extends React.Component<Props, State> {
     );
   }
 
+  _renderArrayChildren(
+    parentVariable: gdVariable,
+    depth: number,
+    origin: VariableOrigin
+  ): Array<React.Node> {
+    return mapFor(0, parentVariable.getChildrenCount(), index => {
+      const variable = parentVariable.getAtIndex(index);
+      return this._renderVariableAndChildrenRows(
+        '' + index,
+        variable,
+        depth + 1,
+        index,
+        parentVariable,
+        origin,
+        /* arrayElement= */ true
+      );
+    });
+  }
+
   _getVariableOrigin = (name: string) => {
     const { variablesContainer, inheritedVariablesContainer } = this.props;
 
+    if (typeof name === 'number') return '';
```

Comment:
Same as before, can we pass an explicit argument rather than relying on variables having multiple types?